### ISSUE TITLE
bug:  IndexError: list index out of range for annotate_text with input string containing a URL with specific type of date 

### ISSUE THREAD

**Issue by jurrriaan** [vmenger/deduce#44]
Using the string below as input for annotate_text gives me an IndexError: list index out of range
```
import deduce
text_input = "https://www.rtl.nl/components/financien/rtlz/2004/06_juni/crucell_sars.pdf"
deduce.annotate_text(text_input)
```

The strings  "https://www.rtl.nl/components/financien/rtlz/2004/crucell_sars.pdf"
 and  "https://www.rtl.nl/components/financien/rtlz/06_juni/crucell_sars.pdf" seem to work however.

**Comment by vmenger** [comment]
Hi @jurrriaan, I can only reproduce it on 1.0.6, the latest version (1.0.8) does not raise an error (although the annotation is not exactly great, but that's a different issue I suppose). Are you able to update?

**Comment by jurrriaan** [comment]
Hi @vmenger, I see I didn't update indeed. I still had 1.0.7. I upgraded to 1.0.8 now but I still find URLs that give the same bug. Could you please try this one?

 ```
import deduce
text_input = "http://mkatan.nl/bnr-radiocolumns/346-9-augustus-afvalpillen" 
deduce.annotate_text(text_input)
```

**Comment by vmenger** [comment]
Yes, now I can reproduce. Spent a little time digging, the problem is that Deduce detects both `6-9` and `9-augustus` as a date, and that *both* of those are contained in a URL. So definitely a bug, albeit a bit of an exotic one.  

At the moment I cannot think of an quick way to solve this specific bug, without imposing additional checks/computations for all Deduce users (and I prefer not to), but we do have some improvements planned that will tackle this issue as well. 

Do other URLs that give this error have the same structure? Not sure about your data/use case, would it be an option to work around it temporarily with:

```python
try:
  deduce.annotate_text(text)
except IndexError:
  deduce.annotate_text(text, dates=False)
```

**Comment by jurrriaan** [comment]
Another URL giving the same error:  "https://www.cultuurconnectie.nl/nieuwsbrieven/2020/nieuwsflits-covid-19-16-7-mei-2020"

The work around is not ideal but good enough to continue for now. Looking forward to an improved version.
Thanks for your help!

**Comment by vmenger** [comment]
@jurrriaan If this issue is still relevant for you, it's now fixed in 2.0.0 (at least, there is a structural solution, and the above examples don't give errors anymore).

In case you want to use the latest version, a small change is needed described here: https://deduce.readthedocs.io/en/latest/migrating.html
